### PR TITLE
Site Settings: Move General section to a separate chunk

### PR DESCRIPTION
**Pull request by tyxla**
We've been working a lot on the Site Settings recently. A lot of stuff has been introduced, enhanced or improved, but this also has some cons: the `settings` chunk is now 1.12MB 😱 

It is obvious that we need to split `settings` to smaller chunks. This has also been discussed in p4TIVU-5Zs-p2 and recommended by @dmsnell and @mtias.

So this PR moves the General settings section to a separate chunk. After this change, the chunk sizes are:

This is not introducing any functional or behavioral changes. But since it's affecting all of the routes and functionality that resides within the General settings section, extensive testing of the General section and its settings will be required.

To test:
* Checkout this branch, or get it going on calypso.live
* Go to `/settings/general/$site`, where `$site` is one of your WordPress.com sites.
* Verify the tab loads correctly with no errors/warnings.
* Change to another tab and come back to General, verify it works like before.
* Play with all settings in the General section and verify they are saved and retrieved correctly.
* Click on the "Delete Site" button and verify it works properly:
  * If you have a Free plan, it should direct you to `/settings/delete-site/$site`
  * If you have a paid plan, it should display a popover to cancel any upgrades first.
* Click on the "Start Over" button and verify it loads `/settings/start-over/$site` correctly.
* Click on the "Theme Setup" button and verify it loads `/settings/theme-setup/$site` correctly.
* If you're on the dev environment (`calypso.localhost`), click on the "Date and Time Format" button and verify it loads `/settings/date-time-format/$site` correctly.
* Try opening any of the above routes in a fresh page session and verify they all load correctly.
* Go to `/settings/general/$site`, where `$site` is one of your connected Jetpack sites.
* Verify the tab loads correctly with no errors/warnings.
* Change to another tab and come back to General, verify it works like before.
* Play with all settings in the General section and verify they are saved and retrieved correctly.
* If you're on the dev environment (`calypso.localhost`), click on the "Date and Time Format" button and verify it loads `/settings/date-time-format/$site` correctly.
* Don't hate me for the long test instructions list 😆

### CODE REVIEW

File: client/my-sites/site-settings/general/main.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+	render() {
+		const {
+			site,
+			sitePurchases,
+			translate,
+		} = this.props;
+
+		return (
+			<Main className="general__main site-settings">
```
Comment (mtias) [inline client/my-sites/site-settings/general/main.jsx]: Can we drop `site-settings`?

**Comment by tyxla** [comment]
@mtias I've renamed `general` to `settings-general` in 7d12e7f. This one could use another review.

File: client/my-sites/site-settings/general/main.jsx (hunk 1 above)
Comment (tyxla) [inline client/my-sites/site-settings/general/main.jsx]: I'd certainly love to, but not right now. It currently contains a lot of the styles within site settings, so I suggest that we leave that for another PR.

**Comment by tyxla** [comment]
This PR has been rebased and ready for a new review.

**Comment by tyxla** [comment]
Just to clarify: tests are failing because linter does not like the fact that I'm introducing usage of `sites-list`. But I'm actually moving some code along that already uses `sites-list`, and removing `sites-list` is out of the scope right now.

**Comment by tyxla** [comment]
This PR has been rebased and is ready for review.

**Comment by tyxla** [comment]
This one has been rebased again. A lot of stuff is happening in settings these days 😓

**Comment by tyxla** [comment]
I've also removed Date and Time from General, now that #12900 has landed.

**Comment by tyxla** [comment]
I've started addressing removing `sites-list` from `General` in #13179; once merged, this PR will be updated accordingly.

**Comment by tyxla** [comment]
I've rebased this, but it'll need some more work to get rid of sites-list before merging. We need to remove the sites-list usages from `DeleteSite` before we continue with this one.

**Comment by tyxla** [comment]
Too many changes are happening here recently, and we can easily lose track of any of them with so many rebases. I'll close this one as we refactor other stuff, and reopen when ready.
